test/library/draft/arrow/Arrow.chpl

```diff
@@ -0,0 +1,253 @@
+module Arrow {
+    require "ArrowDecl.chpl";
+    public use ArrowDecl;
+    public use SysCTypes, CPtr;
+    // -------------------------- Type Declarations and Functions -------------
+
+    type ArrowArray = GArrowArray;
+
+    proc arrowInt32(): GArrowInt32DataType {
+        return garrow_int32_data_type_new();
+    }
+    
+    proc arrowInt64(): GArrowInt32DataType {
+        return garrow_int64_data_type_new();
+    }
+    proc arrowString(): GArrowStringDataType {
+        return garrow_string_data_type_new();
+    }
+    proc arrowBool(): GArrowBooleanDataType {
+        return garrow_boolean_data_type_new();
+    }
+    //--------------------------- Array building functions --------------------
+
+    proc array(array: [?arrDom] ?arrayType, validity: [?validityDom] int = [-1], 
+                specifiedType = nil): GArrowArray {
+        var fakeReturn: GArrowArray;
+        // Sorting based on the specifiedType and calling the appropriate function
+        //if(!specifiedType)
+        select arrayType {
+            when int do return int64Array(array, validity);
+            when string do return stringArray(array, validity);
+            when bool do return boolArray(array, validity);
+            otherwise {
+                writeln ("Unsupported type, \nreturning nil"); 
+                return fakeReturn;
+            }
+        }
+    }
+    
+    proc int64Array(array: [?arrDom] int, validity: [?validityDom] int) : GArrowInt64Array {
+        
+        var retval: GArrowInt64Array;
+        
+        var builder: GArrowInt64ArrayBuilder;
+        var success: gboolean = 1;
+        var error: GErrorPtr;
+
+        builder = garrow_int64_array_builder_new();
+        //writeln("0");
+
+        if (success) {
+            //writeln("1");
+            var intArrLen: gint64 = array.size: gint64;
+            //writeln("1.1 + ", intArrLen, " " , array.size);
+            var intValArr: [0..#intArrLen] gint64;
+            forall i in 0..#intArrLen {
+                intValArr[i] = array[i]: gint64;
+            }
```

Comment:
A cleaner way to write this is: `var intValArr = [val in array] val: gint64;`, which I believe is exactly equivalent to these 6 lines. There may even be a better way to do this, but I couldn't find anything looking briefly. 

Also, do we need to cast those values to `gint64` here? I thought that gint64 was just a type in `ArrayDecl.chpl` that is essentially just an alias for regular int, but I am still getting used to this extern stuff, so I am likely wrong.

Reply:
Thanks for the tip!
I think you're right, the cast is not necessary. I wrote the `int32` function first and thus I had to use the cast, I guess I didn't think about removing the cast for this one.